core-plugins/src/test/java/io/cdap/plugin/batch/aggregator/DedupAggregatorValidationTest.java

```diff
@@ -75,12 +91,47 @@ public void testValidateHappyPathNonNumericFilterField() {
 
     @Test
     public void testValidationExceptionNonNumericFilterField() {
-        functionInfo = new DedupConfig.DedupFunctionInfo("profession", DedupConfig.Function.MIN);
+        functionInfo = new DedupConfig.DedupFunctionInfo("hasInsurance", DedupConfig.Function.MIN);
         Mockito.doReturn(functionInfo).when(dedupConfig).getFilter();
         DedupAggregator dedupAggregator = new DedupAggregator(dedupConfig);
         dedupAggregator.configurePipeline(pipelineConfigurer);
         Assert.assertEquals(collector.getValidationFailures().size(), 1);
         Assert.assertEquals(collector.getValidationFailures().get(0).getMessage(), "Unsupported filter " +
-                "operation MIN(profession): Field 'profession' is non numeric ");
+                "operation MIN(hasInsurance): Field has a type that is not supported for deduplication operations");
+    }
+
+    @Test
+    public void testSelectFilterFunctionFieldDateTime() {
+        aggValue = StructuredRecord.builder(inputSchema).setDateTime("time of purchase",
+                LocalDateTime.of(2015,
+                        Month.JULY, 29, 19, 30, 40)).build();
+        record  = StructuredRecord.builder(inputSchema).setDateTime("time of purchase",
+                LocalDateTime.of(2017,
+                        Month.JULY, 29, 19, 30, 40)).build();
+
+        functionInfo = new DedupConfig.DedupFunctionInfo("time of purchase", DedupConfig.Function.MIN);
+        Mockito.doReturn(uniqueFields).when(dedupConfig).getUniqueFields();
+        Mockito.doReturn(functionInfo).when(dedupConfig).getFilter();
+        DedupAggregator dedupAggregator = new DedupAggregator(dedupConfig);
+        dedupAggregator.initialize(context);
+        StructuredRecord actualRes = dedupAggregator.mergeValues(aggValue, record);
+        Assert.assertEquals(actualRes.getDateTime("time of purchase"), LocalDateTime.of(2015,
+                Month.JULY, 29, 19, 30, 40));
+    }
+
+    @Test
+    public void testSelectFilterFunctionFieldDecimal() {
+        aggValue = StructuredRecord.builder(inputSchema).setDecimal("price",
+                new BigDecimal("4000.89")).build();
+        record  = StructuredRecord.builder(inputSchema).setDecimal("price",
+                new BigDecimal("4000.91")).build();
+
+        functionInfo = new DedupConfig.DedupFunctionInfo("price", DedupConfig.Function.MAX);
+        Mockito.doReturn(uniqueFields).when(dedupConfig).getUniqueFields();
+        Mockito.doReturn(functionInfo).when(dedupConfig).getFilter();
+        DedupAggregator dedupAggregator = new DedupAggregator(dedupConfig);
+        dedupAggregator.initialize(context);
+        StructuredRecord actualRes = dedupAggregator.mergeValues(aggValue, record);
+        Assert.assertEquals(actualRes.getDecimal("price"), new BigDecimal("4000.91"));
```

Comment:
above tests for datetime and decimal specifically.

Reply:
Can you add tests to all other types (timestamps, etc)?

Reply:
added additional tests:)